src/de/value.rs

```diff
@@ -111,27 +111,38 @@ impl de::VariantVisitor for VariantVisitor
         Ok(())
     }
 
-    /// `visit_seq` is called when deserializing a tuple-like variant.
-    fn visit_seq<V>(&mut self, _visitor: V) -> Result<V::Value, Self::Error>
-        where V: de::Visitor
+    fn visit_newtype<T>(&mut self) -> Result<T, Self::Error>
+        where T: de::Deserialize,
     {
-        unimplemented!()
+        if let Some(element) = self.0.take() {
+            if let Content::Text(content) = element.members {
+                de::Deserialize::deserialize(&mut StringDeserializer(Some(content)))
+            }
+            else {
+                Err(Error::SyntaxError(ErrorCode::Expected("non-empty element"), 0, 0))
+            }
+        }
+        else {
```

Comment:
minor nit: change those `else`s to `} else {` or use one of the `Option` -> `Result` conversion functions like `ok_or` or `ok_or_else` instead of an `if`+`else`